teku/src/main/java/tech/pegasys/teku/cli/subcommand/GenesisCommand.java

```diff
@@ -89,6 +101,26 @@ public void generate(
     }
   }
 
+  private ExecutionPayloadHeader createExecutionPayloadHeader(MockGenesisParams genesisParams) {
+    Bytes32 blockHash = Bytes32.fromHexString(genesisParams.eth1BlockHash);
+    return new ExecutionPayloadHeader(
+        Bytes32.ZERO,
+        Bytes20.ZERO,
+        Bytes32.ZERO,
+        Bytes32.ZERO,
+        Bytes.wrap(new byte[SpecConfig.BYTES_PER_LOGS_BLOOM]),
+        blockHash,
+        UInt64.ZERO,
+        UInt64.valueOf(genesisParams.genesisGasLimit),
+        UInt64.ZERO,
+        UInt64.valueOf(genesisParams.genesisTime),
+        Bytes.EMPTY,
+        Bytes32.rightPad(
+            Bytes.ofUnsignedLong(genesisParams.genesisBaseFee, ByteOrder.LITTLE_ENDIAN)),
+        blockHash,
+        Bytes32.ZERO);
```

Comment:
I'm a little surprised we don't wind up needing to get the genesis payload from the EL client.  The genesis state root won't be zero for example.  I suspect if all CL clients use the same values it probably works fine but it could cause confusion if these zero'd values are sent as part of messages to the EL client as they won't match what it expects.

Reply:
TBH I was thinking of passing EL http API address to retrieve the head block and make genesis execution payload from it, but it seemed a bit kind of overcomplicated to have a EL running for generating genesis. 

We discussed with Mikhail about should we pass the full `ExecutionPayloadHeader` for `genesis mock` or just a couple of additional params (`genesisGasLimit` and `genesisBaseFee`) and fill other with zeroes. Other fields don't affect anything than genesis state hash root. So if you are doing standalone mocking you shouldn't care about those fields. If you are doing some kind of interop you may always pick the `genesis.ssz` from the other party.